services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/drs/DrsManagerImpl.java

```diff
@@ -58,7 +88,53 @@ public ServiceInformation getServiceInformation() {
         result.setUpdatedAt(UPDATED_AT);
         result.setEnvironment(stackConfiguration.getStack());
         result.setVersion(stackConfiguration.getStackInstance());
-        result.setUrl(DRS_URL);
+        result.setUrl(baseURL);
+        return result;
+    }
+
+    @Override
+    public DrsObject getDrsObject(final Long userId, final String id)
+            throws NotFoundException, DatastoreException, UnauthorizedException, IllegalArgumentException {
+        final UserInfo userInfo = userManager.getUserInfo(userId);
+        final DrsObject result = new DrsObject();
+        final IdAndVersion idAndVersion = IdAndVersion.parse(id);
+        final Entity entity = entityManager.getEntityForVersion(userInfo, idAndVersion.getId().toString(),
+                idAndVersion.getVersion().orElseThrow(() ->
+                        new NotFoundException(String.format("Drs object id %s does not exists", id))), null);
+
+        result.setId(id);
+        result.setName(entity.getName());
+        final String selfURI = String.format("%s://%s/%s", DRS, REGISTERED_HOSTNAME, id);
+        result.setSelf_uri(selfURI);
+        result.setCreated_time(entity.getCreatedOn());
+        result.setUpdated_time(entity.getModifiedOn());
+        result.setDescription(entity.getDescription());
+
+        if (entity instanceof FileEntity) {
+            final FileEntity file = (FileEntity) entity;
+            final FileHandle fileHandle = fileHandleManager.getRawFileHandle(userInfo, file.getDataFileHandleId());
+            result.setSize(fileHandle.getContentSize());
+            result.setVersion(file.getVersionNumber());
+            result.setMime_type(fileHandle.getContentType());
+            final List<Checksum> checksums = new ArrayList();
+            final Checksum checksum = new Checksum();
+            checksum.setChecksum(fileHandle.getContentMd5());
+            checksum.setType(CHECKSUM_TYPE);
+            checksums.add(checksum);
+            result.setChecksums(checksums);
+            final List<AccessMethod> accessMethods = new ArrayList<>();
+            final String accessId = String.join(DELIMETER, FileHandleAssociateType.FileEntity.name(), id,
```

Comment:
in the future this can be pull into a util that can go both ways data->string and string->data